File: mobile/lib/features/communications/domain/entities/student_communication.dart
class StudentCommunication {
  final String id;
  final String schoolId;
  final String studentId;
  final String title;
  final String body;
  final DateTime? createdDate;

  StudentCommunication({
    required this.id,
    required this.schoolId,
    required this.studentId,
    required this.title,
    required this.body,
    required this.createdDate,
  });
}

class CommunicationNotification {
  final String id;
  final String schoolId;
  final String studentId;
  final String title;
  final String body;
  final bool isRead;
  final DateTime? createdDate;

  CommunicationNotification({
    required this.id,
    required this.schoolId,
    required this.studentId,
    required this.title,
    required this.body,
    required this.isRead,
    required this.createdDate,
  });
}

File: web/features/communications/domain/entities/student-communication.ts
export type TeacherCommunicationCourse = {
  id: string;
  name: string;
};

export type TeacherCommunicationStudent = {
  id: string;
  firstName: string;
  lastName: string;
};

export type StudentCommunication = {
  id: string;
  schoolId: string;
  studentId: string;
  authorUserId: string;
  title: string;
  body: string;
  createdDate: string;
};
